android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -1279,7 +1274,7 @@ public void run()
 							public void progress(int progress)
 							{
 								KrollDict data = new KrollDict();
-								double currentProgress = ((double) progress / totalLength);
+								double currentProgress = ((double) progress / contentLength);
```

Comment:
@ypbnv, we should check if content length is zero. I can see this potentially happening with an HTTP "PUT" without a body.
```java
public void progress(int progress)
{
	if (contentLength <= 0) {
		return;
	}

	// The rest of the code goes here...
}
```

Reply:
Done.